browser/ui/brave_tooltips/brave_tooltip_popup.h

```diff
@@ -0,0 +1,155 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_BRAVE_TOOLTIPS_BRAVE_TOOLTIP_POPUP_H_
+#define BRAVE_BROWSER_UI_BRAVE_TOOLTIPS_BRAVE_TOOLTIP_POPUP_H_
+
+#include <cstdint>
+#include <memory>
+#include <string>
+
+#include "base/scoped_observation.h"
+#include "brave/browser/ui/brave_tooltips/brave_tooltip.h"
+#include "ui/base/metadata/metadata_impl_macros.h"
+#include "ui/display/display_observer.h"
+#include "ui/gfx/animation/animation_delegate.h"
+#include "ui/gfx/shadow_util.h"
+#include "ui/gfx/shadow_value.h"
+#include "ui/views/widget/widget_delegate.h"
+#include "ui/views/widget/widget_observer.h"
+
+class Profile;
+
+namespace gfx {
+class LinearAnimation;
+class Point;
+class Rect;
+class Size;
+}  // namespace gfx
+
+namespace views {
+class Widget;
+}  // namespace views
+
+namespace brave_tooltips {
+
+class BraveTooltipView;
+
+// The widget delegate of a tooltip popup. The view is owned by
+// the widget
+class BraveTooltipPopup : public views::WidgetDelegateView,
+                          public views::WidgetObserver,
+                          public gfx::AnimationDelegate,
+                          public display::DisplayObserver {
+ public:
+  METADATA_HEADER(BraveTooltipPopup);
+
+  BraveTooltipPopup(Profile* profile, std::unique_ptr<BraveTooltip> tooltip);
+  ~BraveTooltipPopup() override;
+
+  // Show the tooltip popup view for the given |profile| and
+  // |tooltip|
```

Comment:
Since this and `Close` are the primary entry points for the functionality here, can we add a small example to the documentation comments?

Reply:
Yes, will do.

Reply:
Added comments to explain the usage.